stdlib/string.gr

```diff
@@ -221,14 +284,17 @@ export let rec charAt = (position, string: String) => {
   let (<) = WasmI32.ltU
   let (==) = WasmI32.eq
   let size = WasmI32.fromGrain(wasmSafeByteLength(string)) >>> 1n
-  let len = WasmI32.fromGrain(wasmSafeLength(string)) >>> 1n
   let position = WasmI32.fromGrain(position) >>> 1n
   let string = WasmI32.fromGrain(string)
   let mut ptr = string + 8n
   let end = ptr + size
   let mut counter = 0n
-  let mut result = 0n
+  let mut result = WasmI32.toGrain(0n): Char
   while (ptr < end) {
+    if (counter == position) {
+      result = tagChar(getCodePoint(ptr))
```

Comment:
This change does appear to change the semantics of this function, FYI

Reply:
I think it just looks like it does—the info about number of bytes isn't needed anymore, so this just saves a load by happening first (and is a bit more correct since we don't potentially load memory we're not supposed to).